From 0d484a353086b4f2d6d5d0f2acfbb52eb610a1a1 Mon Sep 17 00:00:00 2001
From: Po Liu <Po.Liu@nxp.com>
Date: Thu, 4 Apr 2019 13:10:35 +0800
Subject: [PATCH 680/706] mscc: felix tsn switch bug fix

fix the qbv disable cycletime = 0.

Signed-off-by: Po Liu <Po.Liu@nxp.com>
(cherry picked from commit e837be484a2a16735748c77fa3972ed988bcd540)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/net/ethernet/mscc/tsn_switch.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/mscc/tsn_switch.c b/drivers/net/ethernet/mscc/tsn_switch.c
index 4a433460a2e0..5e39a010ccf5 100644
--- a/drivers/net/ethernet/mscc/tsn_switch.c
+++ b/drivers/net/ethernet/mscc/tsn_switch.c
@@ -95,8 +95,9 @@ int switch_qbv_set(struct net_device *ndev, struct tsn_qbv_conf *shaper_config)
 		return -EINVAL;
 	}
 
-	if ((admin_basic->cycle_time < SWITCH_TAS_CT_MIN) ||
-	    (admin_basic->cycle_time > SWITCH_TAS_CT_MAX)) {
+	if (((admin_basic->cycle_time < SWITCH_TAS_CT_MIN) ||
+	    (admin_basic->cycle_time > SWITCH_TAS_CT_MAX)) &&
+	    shaper_config->gate_enabled) {
 		netdev_info(ndev, "Invalid admin_cycle_time %u ns\n",
 			    admin_basic->cycle_time);
 		return -EINVAL;
-- 
2.17.1

